cmd/argocd-util/commands/config.go

```diff
@@ -0,0 +1,194 @@
+				cmdutil.SetParameterOverrides(&app, appOpts.Parameters)
+				cmdutil.SetLabels(&app, labels)
+			}
+			if app.Name == "" {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+
+			appClientset := appfake.NewSimpleClientset(&app)
+			created, err := appClientset.ArgoprojV1alpha1().Applications("argocd").Create(context.Background(), &app, v1.CreateOptions{})
+			errors.CheckError(err)
+
+			errors.CheckError(cmdutil.PrintResource(created, outputFormat))
+		},
+	}
+	command.Flags().StringVar(&appName, "name", "", "A name for the app, ignored if a file is set (DEPRECATED)")
+	command.Flags().StringVarP(&fileURL, "file", "f", "", "Filename or URL to Kubernetes manifests for the app")
+	command.Flags().StringArrayVarP(&labels, "label", "l", []string{}, "Labels to apply to the app")
+	command.Flags().StringVar(&outputFormat, "o", "yaml", "Output format (yaml|json)")
+
+	// Only complete files with appropriate extension.
+	err := command.Flags().SetAnnotation("file", cobra.BashCompFilenameExt, []string{"json", "yaml", "yml"})
+	errors.CheckError(err)
+
+	cmdutil.AddAppFlags(command, &appOpts)
+	return command
+}
+
+// NewGenProjectConfigCommand generates declarative configuration file for given project
+func NewGenProjectConfigCommand() *cobra.Command {
+	var (
+		opts         cmdutil.ProjectOpts
+		fileURL      string
+		outputFormat string
+	)
+	var command = &cobra.Command{
+		Use:   "proj PROJECT",
+		Short: "Generate declarative config for a project",
+		Run: func(c *cobra.Command, args []string) {
+			var proj v1alpha1.AppProject
+			fmt.Printf("EE: %d/%v\n", len(opts.SignatureKeys), opts.SignatureKeys)
+			if fileURL == "-" {
+				// read stdin
+				err := cmdutil.ReadProjFromStdin(&proj)
+				errors.CheckError(err)
+			} else if fileURL != "" {
+				// read uri
+				err := cmdutil.ReadProjFromURI(fileURL, &proj)
+				errors.CheckError(err)
+
+				if len(args) == 1 && args[0] != proj.Name {
+					log.Fatalf("project name '%s' does not match project spec metadata.name '%s'", args[0], proj.Name)
+				}
+			} else {
+				// read arguments
+				if len(args) == 0 {
+					c.HelpFunc()(c, args)
+					os.Exit(1)
+				}
+				projName := args[0]
+				proj = v1alpha1.AppProject{
+					TypeMeta: v1.TypeMeta{
+						Kind:       application.AppProjectKind,
+						APIVersion: application.Group + "/v1aplha1",
+					},
+					ObjectMeta: v1.ObjectMeta{Name: projName},
+					Spec: v1alpha1.AppProjectSpec{
+						Description:       opts.Description,
+						Destinations:      opts.GetDestinations(),
+						SourceRepos:       opts.Sources,
+						SignatureKeys:     opts.GetSignatureKeys(),
+						OrphanedResources: cmdutil.GetOrphanedResourcesSettings(c, opts),
+					},
+				}
+			}
+
+			appClientset := appfake.NewSimpleClientset(&proj)
+			created, err := appClientset.ArgoprojV1alpha1().AppProjects("argocd").Create(context.Background(), &proj, v1.CreateOptions{})
+			errors.CheckError(err)
+			errors.CheckError(cmdutil.PrintResource(created, outputFormat))
```

Comment:
Same as `config app` command this prints `status`, `creationTimestamp` and `namespace`. These fields should not be printed.